belash-bo/lab1-BasicIO/user.c

```diff
@@ -0,0 +1,74 @@
+/******************************************************************************/
+/* Files to Include                                                           */
+/******************************************************************************/
+
+#ifdef __XC32
+    #include <xc.h>          /* Defines special function registers, CP0 regs  */
+#endif
+
+#include <stdint.h>          /* For uint32_t definition                       */
+#include <stdbool.h>         /* For true/false definition                     */
+#include "user.h"            /* variables/params used by user.c               */
+
+/******************************************************************************/
+/* User Functions                                                             */
+/******************************************************************************/
+
+void InitApp(void)
+{
+    // LED output
+    // Disable analog mode
+    ANSELBbits.ANSB9 = 0;
+    // Set direction to output
+    TRISBbits.TRISB9 = 0;
+    // LED off
+    LATBbits.LATB9 = 0;
+    // Button input
+    // Disable analog mode
+    ANSELBbits.ANSB13 = 0;
+    // Set directions to input
+    TRISBbits.TRISB13 = 1;
+}
+
+void Delay(int n) {
+    volatile int i;
+    for (i=0; i<n; i++) {
+    }
+ }
+
+#define VER 3
+
+#if VER==1
+void Flash_LED(void) {
+    while (1) {
```

Comment:
All infinite cycles should be used inside `main()` function unless there are some major reasons to do otherwise, like:
handling exceptions (control flow exits infinite loop if something goes wrong);
halting CPU on erroneous conditions (`while (! condition) LED_PORT |= ALL_LEDS; LED_PORT = 0`)
using software-defined state machines
...